src/datasources/accounts/accounts.datasource.ts

```diff
@@ -192,4 +210,33 @@ export class AccountsDatasource implements IAccountsDatasource, OnModuleInit {
       );
     }
   }
+
+  /**
+   * Checks if the client IP address has reached the account creation rate limit.
+   *
+   * If the client IP address is invalid, a warning is logged.
+   * If the client IP address is valid and rate limit is reached, a {@link LimitReachedError} is thrown.
+   *
+   * @param clientIp - client IP address.
+   */
+  private async checkCreationRateLimit(clientIp: string): Promise<void> {
+    if (!clientIp || !AccountsDatasource.SIMPLE_IPV4_REGEX.test(clientIp)) {
+      this.loggingService.warn(
+        `Invalid client IP while creating account: ${clientIp}`,
+      );
+    } else {
+      const current = await this.cacheService.increment(
+        CacheRouter.getRateLimitCacheKey(
+          `${AccountsDatasource.ACCOUNT_CREATION_CACHE_PREFIX}_${clientIp}`,
+        ),
+        this.accountCreationRateLimitPeriodSeconds,
+      );
+      if (current > this.accountCreationRateLimitCalls) {
+        this.loggingService.warn(
+          `Limit of ${this.accountCreationRateLimitCalls} reached for IP ${clientIp}`,
+        );
```

Comment:
As before, I added a log line here also, as getting too many rate limit conditions shouldn't be normal.